### PR TITLE
[PKG-2140] quantile-forest 1.1.2

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: https://github.com/zillow/quantile-forest/releases
License: https://github.com/zillow/quantile-forest/blob/v1.1.2/LICENSE
Requirements: 
- https://github.com/zillow/quantile-forest/blob/v1.1.2/pyproject.toml
- https://github.com/zillow/quantile-forest/blob/v1.1.2/quantile_forest/_min_dependencies.py
- https://github.com/zillow/quantile-forest/blob/v1.1.2/setup.py

Actions:
1. Update `source/url `because the source code of **v1.1.2** from PyPi is broken: file `quantile_forest/version.txt` cannot be found.
2. Add flags `--no-deps --no-build-isolation` to `script`
3. Update pinnings in `host`
4. Reorder `run` dependencies
5. Add `pytest` to `test/commands` and `test/requires`
6. Add `description` 
7. Add doc and dev urls
8. Add `license_family`

### CODE REVIEW

**Comment by sumit0190** [comment]
There is a single, giant test-file in the `tests` folder that you can run and validate the build.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -6,41 +6,55 @@ package:
...
   host:
     - python
-    - cython >=3.0a4
-    - numpy >=1.23
-    - scipy >=1.4
-    - scikit-learn >=1.0
+    - cython 0.29.33
+    - numpy {{ numpy }}
     - pip
+    - scikit-learn
```
Comment (boldorider4) [inline recipe/meta.yaml]: why don't these host dependencies need exact pinning (i.e. removed >=1.0, >=1.4)?

Comment (skupr-anaconda) [reply recipe/meta.yaml]: We have scikit-learn  with python 3.11 support:
```
scikit-learn                   1.1.3 py310h313beb8_1  pkgs/main
scikit-learn                   1.1.3 py310hc377ac9_0  pkgs/main
scikit-learn                   1.1.3 py311h313beb8_1  pkgs/main
scikit-learn                   1.1.3  py38h313beb8_1  pkgs/main
scikit-learn                   1.1.3  py38hc377ac9_0  pkgs/main
scikit-learn                   1.1.3  py39h313beb8_1  pkgs/main
scikit-learn                   1.1.3  py39hc377ac9_0  pkgs/main
scikit-learn                   1.2.0 py310h313beb8_0  pkgs/main
scikit-learn                   1.2.0 py310h313beb8_1  pkgs/main
scikit-learn                   1.2.0  py38h313beb8_0  pkgs/main
scikit-learn                   1.2.0  py38h313beb8_1  pkgs/main
scikit-learn                   1.2.0  py39h313beb8_0  pkgs/main
scikit-learn                   1.2.0  py39h313beb8_1  pkgs/main
scikit-learn                   1.2.1 py310h313beb8_0  pkgs/main
scikit-learn                   1.2.1 py311h313beb8_0  pkgs/main
scikit-learn                   1.2.1  py38h313beb8_0  pkgs/main
scikit-learn                   1.2.1  py39h313beb8_0  pkgs/main
scikit-learn                   1.2.2 py310h313beb8_0  pkgs/main
scikit-learn                   1.2.2 py310h313beb8_1  pkgs/main
scikit-learn                   1.2.2 py311h313beb8_0  pkgs/main
scikit-learn                   1.2.2 py311h313beb8_1  pkgs/main
scikit-learn                   1.2.2  py38h313beb8_0  pkgs/main
scikit-learn                   1.2.2  py38h313beb8_1  pkgs/main
scikit-learn                   1.2.2  py39h313beb8_0  pkgs/main
scikit-learn                   1.2.2  py39h313beb8_1  pkgs/main
```
But all of the versions have been rebuilt for different reasons. It's hard to decide which version to choose and how they correlate to scipy versions.

**Comment by skupr-anaconda** [comment]
> There is a single, giant test-file in the `tests` folder that you can run and validate the build.

Done

**Review by sumit0190 — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: I've pinned host dependencies:
```
    - scikit-learn 1.1.3
    - scipy 1.9.3
```

Comment (skupr-anaconda) [reply recipe/meta.yaml]: But they failed. So it's not a good idea because they conflict with numpy

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Updated as:
```
    - scikit-learn 1.2.2
    - scipy 1.10.1
```
It works

**Review by Arishamays1 — APPROVED** [review]

**Pull request merged**